.config/nextest.toml

```diff
@@ -7,57 +7,44 @@ fail-fast = false
 failure-output = "immediate-final"
 # Default timeout tests after 5 minutes
 slow-timeout = { period = "60s", terminate-after = 5 }
-# Retry failing tests in order to not block builds on flaky tests
+# Retry failing tests once in order to not block builds on flaky tests
 retries = 1
 
-# isolate slow and flaky tests to run alone (no multi-thread / no other tests running)
-[test-groups]
-no-concurrency = { max-threads = 1 }
+[[profile.default.overrides]]
+# Tests that should run single threaded but with otherwise default settings
+threads-required = "num-test-threads"
+filter = '''
+(package(iota-graphql-rpc) and (binary(e2e_tests) or binary(examples_validation_tests) or test(test_query_cost)))
+or (package(iota-cluster-test) and binary(local_cluster_test))
+or (package(iota-indexer) and binary(ingestion_tests))
+'''
 
 [[profile.default.overrides]]
-# Slow tests get only 1 retry (like the default) but a longer timeout
-test-group = 'no-concurrency'
+# Slow tests that should run single threaded and only get 1 retry (like the default) but a longer timeout
+threads-required = "num-test-threads"
 slow-timeout = { period = "120s", terminate-after = 5 }
 filter = '''
-test(generate_cursor_test)
-or test(verify_latest_tx_replay_testnet_mainnet)
-or test(ensure_no_tombstone_fragmentation_in_stack_frame_after_flush)
-or test(ensure_no_tombstone_fragmentation_in_stack_frame_with_ignore_tombstones)
-or test(test_format)
-or test(test_good_snapshot)
-or test(test_insufficient_balance_will_retry_success)
-or test(test_byzantine_peer_handling)
-or test(test_json_rpc_spec)
-or test(cluster_test)
+(package(iota-cluster-test) and test(cluster_test))
+or (package(iota-core) and (test(ensure_no_tombstone_fragmentation_in_stack_frame_after_flush) or test(ensure_no_tombstone_fragmentation_in_stack_frame_with_ignore_tombstones) or test(test_format)))
+or (package(iota-cost) and test(test_good_snapshot))
+or (package(iota-faucet) and test(test_insufficient_balance_will_retry_success))
+or (package(iota-network) and test(test_byzantine_peer_handling))
+or (package(iota-open-rpc) and test(test_json_rpc_spec))
```

Comment:
Why did the format change?

Reply:
I added the package names, so we not simply filter by the test names, in case two tests have the same name in different packages in the future. This way I also found out that one of the filters was not even working, because the test is in a different workspace. Thats why I also added another nextest config for the external-crates workspace.

Reply:
Oh, I missed that.